lib/hammer_cli_foreman_ansible/associated_ansible_resource.rb

```diff
@@ -0,0 +1,16 @@
+module HammerCLIForemanAnsible
+  module AssociatedAnsibleResource
+    def get_current_ids
+      resources = HammerCLIForeman.record_to_common_format(resource.call(association_name(true), { id: get_identifier }))
+      resources.map do |resource|
+        next resource['id'] unless self.class.command_names.include?('remove')
+
+        if resource['inherited'] && resource['id'] == option_ansible_role_id
+          raise ArgumentError, _('Ansible role %s is inherited and cannot be removed.') % resource['name']
+        end
```

Comment:
This sadly does not catch all the cases when modifying hostgroup's roles. The list of roles for a hostgroup contains:
a) Roles explicitly assigned to the hostgroup
b) Roles assigned to hostgroup's ancestors
c) Roles explicitly assigned to hosts belonging to the hostgroup

Roles in group a) can be modified freely, roles in group b are caught by this check, but group c goes completely unnoticed and signals succesful removal even though nothing happens.

To me the easiest way out would be to treat "inherited" as "not directly assigned" instead of "passed from an ancestor of the same type", although it might be confusing if "inherited" has different meanings across the project. Maybe safer route would be renaming the field from "inherited" to "explicitly assigned" and just negating the value?